core/src/poh_recorder.rs

```diff
@@ -51,6 +51,78 @@ type Result<T> = std::result::Result<T, PohRecorderError>;
 pub type WorkingBankEntry = (Arc<Bank>, (Entry, u64));
 pub type BankStart = (Arc<Bank>, Arc<Instant>);
 
+pub struct Record {
+    pub mixin: Hash,
+    pub transactions: Vec<Transaction>,
+    pub slot: Slot,
+    pub sender: Sender<Result<()>>,
+}
+impl Record {
+    pub fn new(
+        mixin: Hash,
+        transactions: Vec<Transaction>,
+        slot: Slot,
+        sender: Sender<Result<()>>,
+    ) -> Self {
+        Self {
+            mixin,
+            transactions,
+            slot,
+            sender,
+        }
+    }
+}
+
+pub struct Recorder {
+    // shared by all users of PohRecorder
+    pub record_sender: Sender<Record>,
+    // unique to this caller
+    pub result_sender: Sender<Result<()>>,
+    pub result_receiver: Receiver<Result<()>>,
+}
+
+impl Clone for Recorder {
+    fn clone(&self) -> Self {
+        Recorder::new(self.record_sender.clone())
+    }
+}
+
+impl Recorder {
+    pub fn new(record_sender: Sender<Record>) -> Self {
+        let (result_sender, result_receiver) = channel();
+        Recorder {
+            // shared
+            record_sender,
+            // unique to this caller
+            result_sender,
+            result_receiver,
+        }
+    }
+    pub fn record(
+        &self,
+        bank_slot: Slot,
+        mixin: Hash,
+        transactions: Vec<Transaction>,
+    ) -> Result<()> {
+        let res = self.record_sender.send(Record::new(
+            mixin,
+            transactions,
+            bank_slot,
+            self.result_sender.clone(),
+        ));
+        if res.is_err() {
+            return Err(PohRecorderError::MaxHeightReached); // re-use an error that gives appropriate behavior
```

Comment:
I would say: "If the channel is dropped, then the validator is shutting down so return that we are hitting the max tick height to stop transaction processing and flush any transactions in the pipeline."